### PR TITLE
o/assertstate: Implement EnforcedValidationSets helper

### PR DESCRIPTION
**Pull request by stolowski**
Implement a helper for getting ValidationSets object for all currently tracked validation sets in Enforce mode.

### CODE REVIEW

File: overlord/assertstate/validation_set_tracking.go [hunk 1]
```diff
@@ -130,3 +133,42 @@ func ValidationSets(st *state.State) (map[string]*ValidationSetTracking, error)
 	}
 	return vsmap, nil
 }
+
+// EnforcedValidationSets returns ValidationSets object with all currently tracked
+// validation sets in enforcing mode.
```
Comment (pedronis) [inline overlord/assertstate/validation_set_tracking.go:138]: s/in/that are in/ is a bit clearer I think

File: overlord/assertstate/validation_set_tracking_test.go [hunk 2]
```diff
@@ -157,3 +186,71 @@ func (s *validationSetTrackingSuite) TestGet(c *C) {
...
+	c.Assert(assertstate.Add(s.st, vs2), IsNil)
+
+	vs3 := s.mockAssert(c, "baz", "5", "invalid")
+	c.Assert(assertstate.Add(s.st, vs3), IsNil)
+
+	valsets, err := assertstate.EnforcedValidationSets(s.st)
+	c.Assert(err, IsNil)
+
+	// foo and bar are in conflict, use this as an indirect way of checking
+	// that both were added to valsets.
```
Comment (pedronis) [inline overlord/assertstate/validation_set_tracking_test.go:253]: this works but is not very realistic, we should probably leave a TODO/XXX to switch to a test using the Check* methods once they have landed

File: overlord/snapstate/storehelpers.go [hunk 3]
```diff
@@ -34,6 +35,8 @@ import (
 
 var currentSnaps = currentSnapsImpl
 
+var EnforcedValidationSets func(st *state.State) (*snapasserts.ValidationSets, error)
```
Comment (pedronis) [inline overlord/snapstate/storehelpers.go:38]: this probably merits a doc comment in the style of the one for example for AutoRefreshAssertions

**Review by pedronis — APPROVED** [review]
implementation looks good, some comments about test and small things

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**